crates/project/src/project.rs

```diff
@@ -4549,6 +4619,27 @@ impl Project {
         })?
     }
 
+    async fn handle_show_document(
+        this: Entity<Self>,
+        envelope: TypedEnvelope<proto::ShowDocument>,
+        mut cx: AsyncApp,
+    ) -> Result<()> {
+        this.update(&mut cx, |this, cx| {
+            if let Ok(uri) = lsp::Url::parse(&envelope.payload.uri) {
```

Comment:
Should we be ignoring this error in particular? Erroring out on invalid URIs seems reasonable to me; we can still ignore non-file URIs, as there's no way for remote counterpart to handle any other URIs anyways.